nativelib/src/main/resources/scala-native/gc/shared/Parsing.c

```diff
@@ -50,17 +54,52 @@ size_t Parse_Size_Or_Default(const char *str, size_t defaultSizeInBytes) {
     return defaultSizeInBytes;
 }
 
+const char *get_defined_or_env(const char *envName) {
+    if (envName == NULL)
+        return NULL;
+
+    const char *env = getenv(envName);
+    if (env != NULL)
+        return env; // override with env var
+
+// check for defined, compile time values
+#if defined(GC_INITIAL_HEAP_SIZE)
+    if (strcmp(envName, "GC_INITIAL_HEAP_SIZE") == 0) {
+        return VAL_STR(GC_INITIAL_HEAP_SIZE);
+    }
+#endif
+
+#if defined(GC_MAXIMUM_HEAP_SIZE)
+    if (strcmp(envName, "GC_MAXIMUM_HEAP_SIZE") == 0) {
+        return VAL_STR(GC_MAXIMUM_HEAP_SIZE);
+    }
+#endif
+
+#if defined(GC_THREAD_HEAP_BLOCK_SIZE)
+    if (strcmp(envName, "GC_THREAD_HEAP_BLOCK_SIZE") == 0) {
+        return VAL_STR(GC_THREAD_HEAP_BLOCK_SIZE);
+    }
+#endif
+
+    return NULL; // no compile time or runtime value
+}
+
 size_t Parse_Env_Or_Default(const char *envName, size_t defaultSizeInBytes) {
-    return Parse_Size_Or_Default(getenv(envName), defaultSizeInBytes);
+    const char *res = get_defined_or_env(envName);
+#ifdef DEBUG_PRINT
+    printf("%s=%s\n", envName, res);
+    fflush(stdout);
```

Comment:
C just-above-trivia

C stdout is line buffered unless some devo has altered the
default. The previous line ends with a "\n" newline, so it will
flush the buffer and make the fflush() spend cycles but do no output.

In other situations, a paranoid person could do an fsync(stdout)
but that is unnecesary here. 

I think stderr is un-buffered, but would have to check